blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/datastax/DatastaxIO.java

```diff
@@ -0,0 +1,147 @@
+package com.rackspacecloud.blueflood.io.datastax;
+
+import com.datastax.driver.core.*;
+import com.datastax.driver.core.exceptions.NoHostAvailableException;
+import com.datastax.driver.core.policies.DCAwareRoundRobinPolicy;
+import com.datastax.driver.core.policies.TokenAwarePolicy;
+import com.rackspacecloud.blueflood.io.IOConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.InetSocketAddress;
+import java.util.Collection;
+import java.util.Set;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+
+/**
+ * Created by shin4590 on 3/10/16.
+ */
+public class DatastaxIO {
+    private static final DatastaxIO INSTANCE = new DatastaxIO();
+    private static final Logger LOG = LoggerFactory.getLogger(DatastaxIO.class);
+    private static final IOConfig ioconfig = IOConfig.singleton();
+
+    private static  Cluster cluster;
+    private static Session session;
+
+//    private final static Meter hostMeter = utils.Metrics.meter(DatastaxIO.class, "Hosts Connected");
+//    private final static Meter openConnectionsMeter = utils.Metrics.meter(DatastaxIO.class, "Total Open Connections");
+//    private final static Meter inFlightQueriesMeter = utils.Metrics.meter(DatastaxIO.class, "Total InFlight Queries");
+//    private final static Meter trashedConnectionsMeter = utils.Metrics.meter(DatastaxIO.class, "Total Trashed Connections");
+//    private final static Meter maxLoadMeter = utils.Metrics.meter(DatastaxIO.class, "Maximum Load");
+
+    static {
+        connect();
+        monitorConnection();
```

Comment:
I feel like there is too much `static` going on in this class. May be we can structure it differently? Here is what I think are the issues with it. We can discuss tomorrow in detail but this is where I am coming from.

When we try to connect to cassandra during static initialization, if for some reason, there is an exception, the class wont get loaded at all which could lead to NoClassDefFound exception when we are trying to use it?. Thats my theory from my past experiences. I am not 100% sure of this.  

Also in this class, I like some separation from ioconfig. Too much going on during class loading which provides less control to us.